### PR TITLE
Initial implementation of Instructor Modal

### LINKED ISSUE #376: Instructor Modal: Initial view of faculty in modal
* Clicking the edit button in the Intructor column opens an instance of the Mark One modal.
* A state object is created in that modal with a list of the instructors.
* This list is passed down to a table with two columns, which displays the instructor names in the left column, in the order that they are displayed in the courses table.
* The modal header component will have the text: "Edit [course]", e.g. "Edit AC 297r". 
* Save and Cancel buttons will be present in the modal footer component.

### PR DESCRIPTION
**Pull request by jonseitz**
This creates the modal for editing the instructor list assigned to a course instance. At present, it just renders the list, in order, with nonfunctional delete and up/down buttons that will be wired up in future PRs.

Note that I haven't implemented anything related to the unsaved changes  yet, as there's no editing functionality with which to test it.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:

- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [x] I have added tests to cover my changes.

## Priority:
- [x] Normal 
- [ ] High 

## Related Issues:

Fixes #376

### CODE REVIEW

File: src/client/components/pages/Courses/__tests__/InstructorModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        deepStrictEqual(renderedNames, instructorNames);
+      });
+      it('should render a delete button', function () {
+        const deleteButton = renderResult.queryByLabelText(
+          `Remove ${instructorNames[0]} from ${testCourse.catalogNumber}`,
+          { exact: false }
+        );
+        notStrictEqual(deleteButton, null);
+      });
+      it('Should not render any arrow buttons', function () {
```
Comment (natalynyu) [inline src/client/components/pages/Courses/__tests__/InstructorModal.test.tsx:81]: It looks like it does render arrow buttons that are just have no click action/`VARIANT.DEFAULT` colored.

**Review by natalynyu — CHANGES_REQUESTED** [review]
This looks great! I really liked the way you handle the logic of rendering the arrows based off of the indexes and the comprehensiveness of the tests. I just had a few small comments on accessibility and a test description.

I also noticed that the mockup doesn't show the `VARIANT.DEFAULT` buttons. I'm thinking that your approach may be better, since the alt text of the `VARIANT.DEFAULT` button describes that the instructor can't be moved, whereas the mockup doesn't have the arrow at all if the instructor can't be moved in that direction. Just wanted to bring this up as a point in case you have further thoughts on that.

Also I noticed that the ticket #376 says that this will include a state to track whether changes have been made, but maybe this makes more sense to be done later after #370 is merged.

File: src/client/components/pages/Courses/InstructorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    <Modal
+      ariaLabelledBy="edit-instructors-header"
+      closeHandler={closeModal}
+      isVisible={isVisible}
+    >
+      <ModalHeader>
+        <span id="edit-instructors-header">
+          {`Edit Instructors for ${instanceIdentifier}`}
+        </span>
+      </ModalHeader>
```
[the remark this answers is not in the document]
Comment (jonseitz) [reply src/client/components/pages/Courses/InstructorModal.tsx]: Excellent point, those are missing. Added them in 872f695

File: src/client/components/pages/Courses/__tests__/InstructorModal.test.tsx (hunk 1 above)
Comment (jonseitz) [reply src/client/components/pages/Courses/__tests__/InstructorModal.test.tsx]: yeah, those should be cleared. Changed the test descriptions in

**Review by natalynyu — APPROVED** [review]
Approved!

**Pull request merged**